src/cargo/core/compiler/custom_build.rs

```diff
@@ -643,10 +643,14 @@ impl BuildOutput {
                 Ok(line) => line.trim(),
                 Err(..) => continue,
             };
-            let mut iter = line.splitn(2, ':');
+            let mut iter = line.splitn(2, "::");
             if iter.next() != Some("cargo") {
-                // skip this line since it doesn't start with "cargo:"
-                continue;
+                // For backwards compatibility, we also accept lines that start with "cargo:".
+                iter = line.splitn(2, ":");
+                if iter.next() != Some("cargo") {
+                    // skip this line since it doesn't start with "cargo:" or "cargo::".
+                    continue;
+                }
```

Comment:
Side note: I think there are two blockers on producing warnings for the old syntax
- Project's MSRV has to be at least when this was stabilized
- We must have cargo warning levels using #12115

Even then we might want to give it some time.  The big problem is old build-dependencies or build-dependencies supporting a very old MSRV will not be able to update to the new syntax.